File: param_pkg/scripts/test_param.py
#!/usr/bin/env python

# http://wiki.ros.org/rospy/Overview/Parameter%20Server

from __future__ import print_function
import rospy
import sys
from std_msgs.msg import String

def getParam():
    print("test_param.py-6- getParam()")

    angle1 = rospy.get_param("/param_py/angle1")
    print("test_param.py-11- angle1: ", angle1)
    float_ = rospy.get_param("/param_py/float")
    print("test_param.py-13- float_: ", float_)

    # a_string = rospy.get_param("a_string")
    # print("test_param.py-8-a_string: ", a_string)
    # relative_name = rospy.get_param("relative_name")
    # private_param = rospy.get_param('~private_name')
    # default_param = rospy.get_param('default_param', 'default_value')
    # gains = rospy.get_param('gains')
    # p, i, d = gains['p'], gains['i'], gains['d']

def setParam():
    print("test_param.py-7- setParam()")
    # Using rospy and raw python objects
    rospy.set_param('a_string', 'baz')
    rospy.set_param('~private_int', 2)
    rospy.set_param('list_of_floats', [1., 2., 3., 4.])
    rospy.set_param('bool_True', True)
    rospy.set_param('gains', {'p': 1, 'i': 2, 'd': 3})

    # Using rosparam and yaml strings
    # rosparam.set_param('a_string', 'baz')
    # rosparam.set_param('~private_int', '2')
    # rosparam.set_param('list_of_floats', "[1., 2., 3., 4.]")
    # rosparam.set_param('bool_True', "true")
    # rosparam.set_param('gains', "{'p': 1, 'i': 2, 'd': 3}")

def checkParam():
    if rospy.has_param('/param_py/angle1'):
        rospy.delete_param('/param_py/angle1')

if __name__ == '__main__':
    try:
        print("test_param.py-37- main()")
        # setParam()
        getParam()
        checkParam()
    except rospy.ROSInterruptException:
        pass

File: param_pkg/src/test_param.cpp

// http://wiki.ros.org/roscpp/Overview/Parameter%20Server

#include "ros/ros.h"
#include "std_msgs/String.h"
#include <sstream>

void setParam();
void getParam();
void checkParam();

void getParam(){
    ROS_INFO("test_param.cpp- getParam()");
    ros::NodeHandle nh;
    std::string string_;
    float float_;

    if (nh.getParam("/param_cpp/float", float_)){
        ROS_INFO("float: %f", float_);
    }
    nh.param<std::string>("/param_cpp/string", string_, "string_");

    if (ros::param::get("/param_cpp/float", float_)){
        ROS_INFO("float: %f", float_);
    }
    ros::param::param<std::string>("/param_cpp/string", string_, "string_");
}

void setParam(){
    ROS_INFO("test_param.cpp- setParam()");
    ros::NodeHandle nh;

    nh.setParam("/global_param", 5);
    nh.setParam("relative_param", "my_string");
    nh.setParam("bool_param", false);
    // ros::param::set("/global_param", 5);
    // ros::param::set("relative_param", "my_string");
    // ros::param::set("bool_param", false);
}

void checkParam(){
    ROS_INFO("test_param.cpp- checkParam()");
    ros::NodeHandle nh;

    if (nh.hasParam("/param_cpp/string")){
        nh.deleteParam("/param_cpp/string");
    }
    // if (ros::param::has("/param_cpp/string")){
    //     ros::param::del("/param_cpp/string");
    // }
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "test_param");
  ros::NodeHandle n;
  
  setParam();
  getParam();

  ros::spin();
  return 0;
}